common/src/main/cpp/classes/openxr_fb_scene_manager.cpp

```diff
@@ -0,0 +1,390 @@
+			if (openxr_interface.is_valid()) {
+				openxr_interface->connect("session_begun", callable_mp(this, &OpenXRFbSceneManager::_on_openxr_session_begun));
+				openxr_interface->connect("session_stopping", callable_mp(this, &OpenXRFbSceneManager::_on_openxr_session_stopping));
+			}
+
+			xr_origin = Object::cast_to<XROrigin3D>(get_parent());
+			if (xr_origin && auto_create && openxr_interface.is_valid() && openxr_interface->is_initialized()) {
+				create_scene_anchors();
+			}
+		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+			if (openxr_interface.is_valid()) {
+				openxr_interface->disconnect("session_begun", callable_mp(this, &OpenXRFbSceneManager::_on_openxr_session_begun));
+				openxr_interface->disconnect("session_stopping", callable_mp(this, &OpenXRFbSceneManager::_on_openxr_session_stopping));
+			}
+
+			if (xr_origin && anchors_created) {
+				remove_scene_anchors();
+			}
+		} break;
+	}
+}
+
+void OpenXRFbSceneManager::_on_openxr_session_begun() {
+	if (xr_origin && auto_create && !anchors_created) {
+		create_scene_anchors();
+	}
+}
+
+void OpenXRFbSceneManager::_on_openxr_session_stopping() {
+	if (anchors_created) {
+		remove_scene_anchors();
+	}
+}
+
+PackedStringArray OpenXRFbSceneManager::_get_configuration_warnings() const {
+	PackedStringArray warnings = Node::_get_configuration_warnings();
+
+	if (is_inside_tree()) {
+		XROrigin3D *origin = Object::cast_to<XROrigin3D>(get_parent());
+		if (origin == nullptr) {
+			warnings.push_back("Must be a child of XROrigin3D");
+		}
+	}
+
+	return warnings;
+}
+
+void OpenXRFbSceneManager::set_default_scene(const Ref<PackedScene> &p_default_scene) {
+	default_scene = p_default_scene;
+}
+
+Ref<PackedScene> OpenXRFbSceneManager::get_default_scene() const {
+	return default_scene;
+}
+
+void OpenXRFbSceneManager::set_scene_setup_method(const StringName &p_method) {
+	scene_setup_method = p_method;
+}
+
+StringName OpenXRFbSceneManager::get_scene_setup_method() const {
+	return scene_setup_method;
+}
+
+void OpenXRFbSceneManager::set_auto_create(bool p_auto_create) {
+	auto_create = p_auto_create;
+}
+
+bool OpenXRFbSceneManager::get_auto_create() const {
+	return auto_create;
+}
+
+void OpenXRFbSceneManager::set_visible(bool p_visible) {
+	visible = p_visible;
+
+	for (KeyValue<StringName, Anchor> &E : anchors) {
+		Node3D *node = Object::cast_to<Node3D>(ObjectDB::get_instance(E.value.node));
+		ERR_CONTINUE_MSG(!node, vformat("Cannot find node for anchor %s.", E.key));
+		if (node) {
+			node->set_visible(p_visible);
+		}
+	}
+}
+
+bool OpenXRFbSceneManager::get_visible() const {
+	return visible;
+}
+
+void OpenXRFbSceneManager::show() {
+	set_visible(true);
+}
+
+void OpenXRFbSceneManager::hide() {
+	set_visible(false);
+}
+
+Error OpenXRFbSceneManager::create_scene_anchors() {
+	ERR_FAIL_COND_V(anchors_created, FAILED);
```

Comment:
Wondering if there's another error we can use to signify the anchors have already been created.

Reply:
Ah, yeah, we could use `ERR_ALREADY_EXISTS`! Updated in my latest push